### PR TITLE
feat: use kubo-rpc-client

### PR DESCRIPTION
**Pull request by SgtPooki**
Tracking issue: https://github.com/ipfs/js-kubo-rpc-client/issues/37
Parent issue: https://github.com/ipfs/js-kubo-rpc-client/issues/6
Originating issue: https://github.com/ipfs/kubo/issues/9125

This PR is intended to update examples so that users will be pointed to `kubo-rpc-client` instead of `ipfs-http-client` when using Javascript/TypeScript to talk to Kubo.

fixes https://github.com/ipfs/js-kubo-rpc-client/issues/37

Things for reviewers to check:

1. Confirm I didn't miss any areas that should be updated (old mentions that should be updated that aren't yet)
2. Confirm the added information is accurate and desired
3. Confirm that the language used is aligned with maintainers' intentions

### CODE REVIEW

**Comment by SgtPooki** [comment]
@lidel @achingbrain need workflow approvals to run here to see if there's any additional changes required to get things passing. There shouldn't be, since interop is passing, but I won't know till these run

**Comment by SgtPooki** [comment]
@hugomrdias can I get permisison to this repo so I can restart workflows in order to better troubleshoot these errors?

cc @tinytb

**Comment by hugomrdias** [comment]
> @hugomrdias can I get permisison to this repo so I can restart workflows in order to better troubleshoot these errors?
> 
> cc @tinytb

sent invite

**Comment by SgtPooki** [comment]
### example-browser-service-worker succeeds locally

```bash
╰─ ✔ ❯ pwd
/Users/sgtpooki/code/work/protocol.ai/ipfs-examples/js-ipfs-examples/examples/browser-service-worker

╰─ ✔ ❯ git log -1 | cat
commit 2b77d8d42d40d579832948c6d54fb4af82e69bf2
Author: Russell Dempsey <1173416+SgtPooki@users.noreply.github.com>
Date:   Thu Oct 6 11:51:13 2022 -0700

    feat: use kubo-rpc-client
    
    fixes https://github.com/ipfs/js-kubo-rpc-client/issues/37
    
╰─ ✔ ❯ gh-issue-info
ProductName:    macOS
ProductVersion: 12.6
BuildVersion:   21G115
Arch: arm64
NodeJS: v16.17.1
npm: 8.15.0
yarn: 1.22.19

╰─ ✔ ❯ npm run test                              8.51   27.9G   23%   100%  ─╯

> example-browser-service-worker@1.0.0 test
> npm run build && playwright test tests


> example-browser-service-worker@1.0.0 build
> webpack --env production

assets by path *.js 1.58 MiB
  assets by chunk 1.56 MiB (id hint: vendors)
    asset 678.js 1.45 MiB [compared for emit] [minimized] [big] (id hint: vendors) 1 related asset
    asset 740.js 91.7 KiB [compared for emit] [minimized] (id hint: vendors) 1 related asset
    asset 94.js 19.5 KiB [compared for emit] [minimized] (id hint: vendors)
  asset 191.js 9.87 KiB [compared for emit] [minimized]
  asset main.js 6.48 KiB [compared for emit] [minimized] (name: main)
  asset 542.js 5.55 KiB [compared for emit] [minimized]
asset favicon.ico 33.7 KiB [compared for emit]
asset index.html 797 bytes [compared for emit]
orphan modules 2.66 MiB [orphan] 874 modules
runtime modules 9.1 KiB 27 modules
cacheable modules 4.47 MiB
  modules by path ../../node_modules/ 4.43 MiB 316 modules
  modules by path ./src/ 37.3 KiB
    ./src/main.js + 1 modules 5.23 KiB [built] [code generated]
    ./src/service.js + 1 modules 22.7 KiB [built] [code generated]
    ./src/worker.js 7.74 KiB [built] [code generated]
    ../../node_modules/css-loader/dist/cjs.js!./src/app.css 1.57 KiB [built] [code generated]
  modules by path ipfs-utils/src/files/*.js (ignored) 30 bytes
    ipfs-utils/src/files/glob-source.js (ignored) 15 bytes [built] [code generated]
    ipfs-utils/src/files/glob-source.js (ignored) 15 bytes [built] [code generated]
  + 7 modules
webpack 5.74.0 compiled successfully in 9585 ms

Running 1 test using 1 worker

  ✓  1 …browser service worker: › should properly load the content of an IPFS hash (393ms)
Starting server...
> Ready on localhost:3000!
Stopping servers...
Servers stopped
```

**Comment by SgtPooki** [comment]
https://github.com/ipfs-examples/js-ipfs-examples/pull/517/commits/770401813d56112bd7ecd7701e2dbef58ccd6c77 should address failure at https://github.com/ipfs-examples/js-ipfs-examples/actions/runs/3252329505/jobs/5338787594#step:6:26

**Review by hacdias — APPROVED** [review]

File: examples/http-client-browser-pubsub/tests/test.js [hunk 1]
```diff
@@ -2,21 +2,23 @@ import { test } from '@playwright/test';
 import { playwright } from 'test-util-ipfs-example'
 import * as ipfsModule from 'ipfs'
 import * as ipfsHttpModule from 'ipfs-http-client'
```
Comment (2color) [inline examples/http-client-browser-pubsub/tests/test.js:4]: Any reason you're still leaving the import here?

File: examples/http-client-browser-pubsub/tests/test.js [hunk 2]
```diff
@@ -2,21 +2,23 @@ import { test } from '@playwright/test';
...
 // Setup
 const play = test.extend({
   ...playwright.servers(),
   ...playwright.daemons(
     {
-      ipfsHttpModule,
     },
     {
       js: {
+        ipfsHttpModule,
```
Comment (SgtPooki) [inline examples/http-client-browser-pubsub/tests/test.js:16]: @2color it's still being used here.

File: examples/http-client-browser-pubsub/tests/test.js (hunk 1 above)
Comment (SgtPooki) [reply examples/http-client-browser-pubsub/tests/test.js:4]: pinged you on https://github.com/ipfs-examples/js-ipfs-examples/pull/517/files/770401813d56112bd7ecd7701e2dbef58ccd6c77#r1021933665

**Comment by SgtPooki** [comment]
merged due to hacdias approval

**Pull request merged**